var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -0,0 +1,153 @@
+                    'em_squall2d_x', 'em_squall2d_y', 'em_grav2d_x',
+                    'em_seabreeze2d_x', 'em_scm_xy'))
+
+    variant('pnetcdf', default=True,
+            description='Parallel IO support through Pnetcdf libray')
+
+    # These patches deal with netcdf & netcdf-fortran being two diff things
+    # Patches are based on:
+    # https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/w/WRF/WRF-3.5_netCDF-Fortran_separate_path.patch
+    patch('patches/4.0/arch.Config.pl.patch', when='@4.0')
+    patch('patches/4.0/arch.configure.defaults.patch', when='@4.0')
+    patch('patches/4.0/arch.conf_tokens.patch', when='@4.0')
+    patch('patches/4.0/arch.postamble.patch', when='@4.0')
+    patch('patches/4.0/configure.patch', when='@4.0')
+    patch('patches/4.0/external.io_netcdf.makefile.patch', when='@4.0')
+    patch('patches/4.0/Makefile.patch', when='@4.0')
+
+    patch('patches/4.2/arch.Config.pl.patch', when='@4.2')
+    patch('patches/4.2/arch.configure.defaults.patch', when='@4.2')
+    patch('patches/4.2/arch.conf_tokens.patch', when='@4.2')
+    patch('patches/4.2/arch.postamble.patch', when='@4.2')
+    patch('patches/4.2/configure.patch', when='@4.2')
+    patch('patches/4.2/external.io_netcdf.makefile.patch', when='@4.2')
+    patch('patches/4.2/var.gen_be.Makefile.patch', when='@4.2')
+    patch('patches/4.2/Makefile.patch', when='@4.2')
+
+    depends_on('mpi')
+    # According to:
+    # http://www2.mmm.ucar.edu/wrf/users/docs/user_guide_v4/v4.0/users_guide_chap2.html#_Required_Compilers_and_1
+    # Section: "Required/Optional Libraries to Download"
+    depends_on('parallel-netcdf', when='+pnetcdf')
+    depends_on('netcdf-c+parallel-netcdf')
+    depends_on('netcdf-fortran')
+    depends_on('jasper')
+    depends_on('libpng')
+    depends_on('zlib')
+    depends_on('perl')
+    # not sure if +fortran is required, but seems like a good idea
+    depends_on('hdf5+fortran+hl+mpi')
+    # build script use csh
+    depends_on('tcsh', type=('build'))
+    # time is not installed on all systems b/c bash provides it
+    # this fixes that for csh install scripts
+    depends_on('time', type=('build'))
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('m4', type='build')
+    depends_on('libtool', type='build')
+
+    def setup_build_environment(self, spack_env):
+        spack_env.set('NETCDF', self.spec['netcdf-c'].prefix)
+        if '+pnetcdf' in self.spec:
+            spack_env.set('PNETCDF', self.spec['parallel-netcdf'].prefix)
+        # This gets used via the applied patch files
+        spack_env.set('NETCDFF', self.spec['netcdf-fortran'].prefix)
+        spack_env.set('PHDF5', self.spec['hdf5'].prefix)
+        spack_env.set('JASPERINC', self.spec['jasper'].prefix.include)
+        spack_env.set('JASPERLIB', self.spec['jasper'].prefix.lib)
+
+        spack_env.prepend_path('PATH', dirname(self.compiler.fc))
```

Comment:
I'm wary about this line. Spack packages should always build with Spack's compiler wrappers so that dependencies get automatically injected into the compile line. `self.compiler.fc` is the actual compiler, not the compiler wrapper. The compiler wrapper should already be in the `PATH`. Is this line actually necessary?

Reply:
@adamjstewart , I removed this line in the latest commit. It compiled with no issues on our cluster. That being said, in 2e3793b84b243087687009d274fda640705eb142 it was claimed that pgf90 was not found without this line.
FYI @bebosudo